### PR TITLE
[build] Reorganize the render tests and collect baselines

### PR DESCRIPTION
**Pull request by tmpsantos**
Move everything that might collect metrics to the metrics/ directory, the only test manifests inside the render-test/ directory are the unit tests of the probes.

Also, we need a sustainable way for mass rebaselining our metrics, in case we make improvements across the board and need to change results for all the platforms.

### CODE REVIEW

File: metrics/next-linux-gcc8-release-metrics.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        "expectations/platform-all"
+    ],
+    "ignore_paths": [
+        "ignores/platform-all.json",
+        "ignores/platform-linux.json"
+    ],
+    "metric_path": "next-linux-gcc8-release",
+    "probes": [
+        "probeGFX",
+        "probeMemory",
```
Comment (alexshalamov) [inline metrics/next-linux-gcc8-release-metrics.json]: Memory probe for small render tests, would be flaky and trigger metric regeneration all the time, same on android bot. That's why we had [real-world](https://github.com/mapbox/mapbox-gl-native/pull/16052/commits/e91a87d777e0f8ffddc1cd43b7777dc50f23f909#diff-65b1263b1c34cc46f4ed7029c15c3425L10) filter.

File: render-test/render_test.cpp [hunk 2]
```diff
@@ -177,9 +178,11 @@ int runRenderTests(int argc, char** argv, std::function<void()> testStatus) {
             }
         }
 
-        bool errored = !metadata.errorMessage.empty() || !runner.run(metadata);
-        bool passed =
-            !errored && (!metadata.outputsImage || !metadata.diff.empty()) && metadata.difference <= metadata.allowed;
+        auto results = runner.run(metadata);
```
Comment (alexshalamov) [inline render-test/render_test.cpp]: This changes previous behavior.  I think that if metadata parsing has failed, test should not be run, e.g.,
```diff
-        auto results = runner.run(metadata);
-
-        bool errored = !metadata.errorMessage.empty() || !(results.probeFailed || results.renderFailed);
+        TestRunner::TestResults results{};
+        bool errored = !metadata.errorMessage.empty();
+        if (!errored) {
+            results = runner.run(metadata);
+            errored = !(results.probeFailed || results.renderFailed);
+        }
```

File: metrics/next-android-render-test-runner-probe-memory.json [hunk 3]
```diff
@@ -1,11 +0,0 @@
-{
-    "base_test_path":"mapbox-gl-js/test/integration",
-    "expectation_paths":["render-test/expected", "render-test/expected/android"],
-    "ignore_paths":["render-test/ignores/ignores.json", "render-test/ignores/linux-ignores.json", "render-test/ignores/android-temp-fail.json"],
-    "vendor_path":"vendor",
-    "asset_path": "mapbox-gl-js/test/integration",
-    "result_path":"/sdcard/",
-    "probes": ["probeMemory"],
-    "metric_path":"metrics/next-android-render-test-runner",
-    "filter": "real-world"
```
Comment (alexshalamov) [inline metrics/next-android-render-test-runner-probe-memory.json]: This removes memory metrics for real-world styles, would we add new memory metrics test-suite in follow-up PRs?

Comment (tmpsantos) [reply metrics/next-android-render-test-runner-probe-memory.json]: Yes, that is the idea. I will add them back for all the platforms with metrics.

**Pull request merged**